kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/swagger/SwaggerIntegrationTest.java

```diff
@@ -53,39 +57,49 @@ public static void closeHttpClient() {
 
     @Test
     public void testSwaggerDocs() throws Exception {
-        String html = invokeGet(getContextRoot()+"docs/");
+        Response response = invokeGet(getContextRoot(3)+"docs/");
         
-        assertThat(Jsoup.parse(html).title()).isEqualTo("Execution Server Documentation");
+        if (response.getStatus()!=200) {
+            //Springboot Swagger Docs is located in other URL
+            response = invokeGet(getContextRoot(1)+"api-docs?url="+getContextRoot(1)+"swagger.json");
+        }
+
+        assertResponse(response);
+        assertThat(Jsoup.parse(responseStr).title()).isIn("Execution Server Documentation", "Swagger UI");
     }
 
     @Test
     @SuppressWarnings("unchecked")
     public void testSwaggerJson() throws Exception {
-        String responseStr = invokeGet(getContextRoot()+"services/rest/server/swagger.json");
-        
+        Response response = invokeGet(TestConfig.getKieServerHttpUrl()+"/swagger.json");
+        assertResponse(response);
+
         ObjectMapper om = new ObjectMapper();
         HashMap<String, Object> hm = (HashMap<String, Object>) om.readValue(responseStr, HashMap.class);
         assertNotNull(hm.get("swagger"));
         assertNotNull(hm.get("info"));
         assertEquals("KIE Server", ((HashMap<String, Object>) hm.get("info")).get("title"));
      }
 
-    protected String getContextRoot() {
-        //Navigate to parent path 3 times to get context root
+    protected String getContextRoot(int foldersUp) {
+        //Navigate to parent path N folders up to get context root
         String url = TestConfig.getKieServerHttpUrl();
         int pos = url.length();
-        for (int i = 0; i < 3; i++) {
+        for (int i = 0; i < foldersUp; i++) {
             pos = url.lastIndexOf('/', pos - 1);
         }
         return url.substring(0, pos + 1);
     }
 
-    protected String invokeGet(String docsUri) {
+    protected Response invokeGet(String docsUri) {
+        logger.info("[GET] " + docsUri);
         WebTarget clientRequest = httpClient.target(docsUri);
-        Response response = clientRequest.request().get();
+        return clientRequest.request().get();
+    }
 
+    private void assertResponse(Response response) {
         assertEquals(200, response.getStatus());
         assertNotNull(response.getEntity());
-        return response.readEntity(String.class);
+        responseStr = response.readEntity(String.class);
```

Comment:
Can you please remove this line from assertResponse and place it for example directly to test methods?
It is a side effect and is unrelated to meaning of assertResponse method - according to its name I would expect it just to check response, not setting some method variable.

Reply:
Done, moved to upper level